apps/rush-lib/src/api/PackageJsonEditor.ts

```diff
@@ -167,6 +174,13 @@ export class PackageJsonEditor {
     return [...this._devDependencies.values()];
   }
 
+  /**
+   * This field is a Yarn-specific feature that allows overriding of package resolution.
+   */
+  public get resolutions(): { [name: string]: string } {
```

Comment:
`ReadonlyArray<PackageJsonDependency>` if that type can be supported, otherwise return the a readonly object

Reply:
The structure is `Record<string, string>` basically the same as `"dependencies"` and `"devDependenices"`, with the only difference being that the name can include glob-like patterns.  I think we can model it the same way.

Reply:
Fixed